unified-scheduler-pool/src/lib.rs

```diff
@@ -194,34 +199,422 @@ impl TaskHandler for DefaultTaskHandler {
+
+                                    // so, we're NOT scheduling at all here; rather, just execute
+                                    // tx straight off. the inter-tx locking deps aren't needed to
+                                    // be resolved in the case of single-threaded FIFO like this.
+                                    active_task_count = active_task_count.checked_add(1).unwrap();
+                                    runnable_task_sender
+                                        .send(ChainedChannel::Payload(task))
+                                        .unwrap();
+                                }
+                                NewTaskPayload::OpenSubchannel(context) => {
+                                    // signal about new SchedulingContext to both handler and
+                                    // accumulator threads
+                                    Self::propagate_context_to_handler_threads(
+                                        &mut runnable_task_sender,
+                                        context,
+                                        handler_count
+                                    );
+                                    executed_task_sender
+                                        .send(ExecutedTaskPayload::OpenSubchannel(()))
+                                        .unwrap();
+                                }
+                                NewTaskPayload::CloseSubchannel => {
+                                    assert!(!session_ending);
+                                    session_ending = true;
+                                }
+                            }
+                        },
+                    };
+
+                    // a really simplistic termination condition, which only works under the
+                    // assumption of single handler thread...
+                    is_finished = session_ending && active_task_count == 0;
+                }
+
+                if session_ending {
+                    executed_task_sender
+                        .send(ExecutedTaskPayload::CloseSubchannel)
+                        .unwrap();
+                    session_result_sender
+                        .send(Some(
+                            accumulated_result_receiver
+                                .recv()
+                                .unwrap()
+                                .unwrap_or_else(initialized_result_with_timings),
+                        ))
+                        .unwrap();
+                    session_ending = false;
+                }
+            }
+        };
+
+        let handler_main_loop = || {
+            let pool = self.pool.clone();
+            let mut bank = context.bank().clone();
+            let mut runnable_task_receiver = runnable_task_receiver.clone();
+            let finished_task_sender = finished_task_sender.clone();
+
+            move || loop {
+                let (task, sender) = select! {
```

Comment:
there's no selection going on here, since there's only one channel receiving?

Reply:
oh, nice question. `select!` here is intentional because the handler will soon use prioritized queue, which will be composed of two channels: 1. low priority tasks (not conflicting with others), 2. high priority tasks (conflicting with others).

so, I'll leave this rather odd select! as-is in this pr.

Reply:
> the handler will soon use prioritized queue, ....

_soon_ :trollface: https://github.com/anza-xyz/agave/pull/627#discussion_r1554626991